ls8/cpu.c

```diff
@@ -2,24 +2,48 @@
 
 #define DATA_LEN 6
 
+#define LDI 0b10000010
+#define NOP 0b00000000
+#define PRN 0b01000111
+#define HLT 0b00000001
+#define STR 0b00001000 // store the next value to reg
+
+// reading whats in ram
+int cpu_ram_read(struct cpu *cpu, int index)
+{
+  return cpu->ram[index];
+}
+
 /**
  * Load the binary bytes from a .ls8 source file into a RAM array
  */
 void cpu_load(struct cpu *cpu)
```

Comment:
You can change ```void cpu_load(char *filename, struct cpu *cpu)``` cpu_load signature to accept two arguments, can read ```print8.ls8``` file passed as argument from ```ls8.c```